airflow/utils/net.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from airflow.configuration import (conf, AirflowConfigException)
+import socket
+
+
+_sentinel = object()
+
+
+def get_hostname(default=None):
+    """
+    A replacement for `socket.getfqdn` that allows configuration to override it's value.
+
+    :param callable|str default: Default if config does not specify. If a callable is
+    given it will be called.
+    """
+
+    hostname = _sentinel
+
+    try:
+        hostname = conf.get('core', 'hostname', fallback=_sentinel)
```

Comment:
@thekashifmalik this makes snese to me: this allows optionally hard-code the hostname in airflow.cfg file.  
What i'm not 100% clear on is how I can overwrite the default value (string or callable).  Could you provide an example? Thanks!

Reply:
You are correct in that this function will not let you override the hostname if a configuration value is set.

Is that an important use case?

Reply:
Sorry, let me clarify: in the case when the configuration value is _not_ set, how do I overwrite the default value to a callable (i.e. I want to implement `socket.gethostname()`, instead of `socket.getfqdn()` )

Reply:
Oh.

You'd do it like this:
```
hostname = get_hostname(socket.gethostname)
```

or 

```
hostname = get_hostname(lambda: socket.gethostname())
```

Reply:
Oh hmm get_hostname() is called in the source code everywhere, so that means I 'd have to modify the source code? or is that something I can do in airflow.cfg? I'm not very familiar with the ConfigParser class. I did tried to modify the hostname variable to a string 'airflow', and got the following error:
```
[2018-02-14 17:38:19 -0800] [93104] [ERROR] Error handling request /admin/
Traceback (most recent call last):
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/gunicorn/workers/sync.py", line 176, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/werkzeug/wsgi.py", line 826, in __call__
    return app(environ, start_response)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/flask/app.py", line 1985, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/flask/app.py", line 1547, in handle_exception
    return self.finalize_request(handler(e), from_error_handler=True)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/apache_airflow-1.10.0.dev0+incubating-py2.7.egg/airflow/www/views.py", line 658, in show_traceback
    hostname=get_hostname(),
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/apache_airflow-1.10.0.dev0+incubating-py2.7.egg/airflow/utils/net.py", line 33, in get_hostname
    hostname = conf.get('core', 'hostname', fallback=_sentinel)
  File "/Users/joy/dev/airflow/venv/lib/python2.7/site-packages/apache_airflow-1.10.0.dev0+incubating-py2.7.egg/airflow/configuration.py", line 202, in get
    ConfigParser.get(self, section, key, **kwargs))
TypeError: get() got an unexpected keyword argument 'fallback'
127.0.0.1 - - [14/Feb/2018:17:38:19 -0800] "GET /admin/ HTTP/1.1" 500 - "-" "-"
```

Reply:
Thinking more about this (I inherited this PR) I think we might want to remove the `default` parameter. it looks like the core concern here is to be able to set a hostname in `airflow.cfg` that overrides whatever `socker.getfqdn` returns.

Reply:
Hmmm, that actually looks like a bug in the function. Let me take a look at the ConfigParser class.